### PR TITLE
ADD fips-assessments for openshift serverless

### PR DESCRIPTION
**Pull request by acmenezes**
Adding fips_assessment.yml with tasks for assessing fips enabled clusters with openshift-serverless.

This PR will include: 
- one task file reference pointing to the fips-assessments repo
- some vars examples on how to enable the tests
- a few changes to gitignore preventing logs and artifacts to be uploaded to this repo

### CODE REVIEW

File: fips_assessment.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          opdev: opcap
+    state: present
+- name: Get Conditions
+  kubernetes.core.k8s_info:
+    namespace: knative-serving
+    name: knative-serving
+    api_version: operator.knative.dev/v1beta1
+    kind: knativeserving
+  register: knative_serving
+  until: knative_serving.resources[0].status.conditions is defined
```
Comment (bcrochet) [inline fips_assessment.yml:20]: Check that `wait_condition` from k8s_info wouldn't work for both this task and the debug that comes next?

File: .gitignore [hunk 2]
```diff
@@ -17,3 +17,12 @@
 # for local dev vars file
 vars.yaml
 vars.yml
+
```
Comment (bcrochet) [inline .gitignore:20]: These changes could go into a separate PR.

**Review by bcrochet — CHANGES_REQUESTED** [review]
Is this supposed to work with all of the FIPS checks?

File: .gitignore (hunk 2 above)
Comment (acmenezes) [reply .gitignore:20]: Ok. I'll separate them.

File: fips_assessment.yml (hunk 1 above)
Comment (acmenezes) [reply fips_assessment.yml:20]: Yeah, I was in the middle of solving that. I'll try the wait condition.

**Comment by acmenezes** [comment]
> Is this supposed to work with all of the FIPS checks?

Yes. The idea is to create a pattern for all FIPS checks. I'm trying to solve for one first. Some of those files you see here will be moved to the fips-assessments repo.

**Comment by acmenezes** [comment]
@bcrochet this one is ready for review. I guess now we can merge it. If you want to test it we need to use the `version` keyword on the git pull in fips-assessment and for now use the ocp-serverless branch to get the files that are there. I'm working in pushing a PR to merge that to master in fips-assessment repo.

File: fips_assessment.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+# Checkout fips-assessments repo for testing tasks
+- name: git checkout fips-assessment tests
+  ansible.builtin.git:
+    repo: https://github.com/opdev/fips-assessments.git
+    dest: /tmp/fips-assessments/
```
Comment (bcrochet) [inline fips_assessment.yml:6]: Does this ever get cleaned up? We should not leave any artifacts behind.

File: vars.yml.fips.example [hunk 4]
```diff
@@ -0,0 +1,17 @@
+auditplan: ["operatorinstall"]
```
Comment (bcrochet) [inline vars.yml.fips.example:1]: Is a separate example necessary?

File: fips_assessment.yml (hunk 3 above)
Comment (acmenezes) [reply fips_assessment.yml:6]: I thought we're supposed to run only with ansible-navigator. Being a container image it gets cleaned up. But you're right if someone runs locally it will remain there.

File: vars.yml.fips.example (hunk 4 above)
Comment (acmenezes) [reply vars.yml.fips.example:1]: Maybe not. I'll try to unify those and let the options commented.

File: fips_assessment.yml (hunk 3 above)
Comment (acmenezes) [reply fips_assessment.yml:6]: I'll add a clean up task for this.

File: vars.yml.fips.example (hunk 4 above)
Comment (acmenezes) [reply vars.yml.fips.example]: Done!

File: fips_assessment.yml (hunk 3 above)
Comment (acmenezes) [reply fips_assessment.yml:6]: Done!

Comment (bcrochet) [reply fips_assessment.yml:6]: I still use ansible-playbook directly, so... :)

File: fips_assessment.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+# Include the path for each fips assessment test below
+# Make sure your tests have proper cleanup tasks
+- name: openshift serverless fips assessment
+  ansible.builtin.include_tasks: /tmp/fips-assessments/openshift-serverless/tasks/main.yml
+
+# Clean up task files
+- name: delete temporary files from fips-assessment
+  ansible.builtin.file:
+    path: /tmp/fips-assessments/
+    state: absent
```
Comment (bcrochet) [inline fips_assessment.yml:17]: New line.

File: vars.yml.example [hunk 6]
```diff
@@ -9,3 +9,7 @@ detailedreports: false
 catalog_source: certified-operators
 catalog_source_namespace: openshift-marketplace
 cleanup: true
+# To run operator tests on FIPS enabled clusters
+#  1. list only packages that have corresponding tests on packages var
+#  2. uncomment and set fips_assessment to true below
+# fips_assessment: true
```
Comment (bcrochet) [inline vars.yml.example:15]: New line.

File: fips_assessment.yml (hunk 5 above)
Comment (acmenezes) [reply fips_assessment.yml]: Done!

File: vars.yml.example (hunk 6 above)
Comment (acmenezes) [reply vars.yml.example]: Done!

**Comment by acmenezes** [comment]
@mrhillsman not sure about the label you put here. I guess we don't need squash. I just have one commit.

File: fips_assessment.yml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
```
Comment (bcrochet) [inline fips_assessment.yml:18]: Looks like L18 has extra whitespace?

Comment (acmenezes) [reply fips_assessment.yml]: I saw it. Removed.

File: .vscode/settings.json [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
+    "ansible.python.interpreterPath": "/bin/python"
+}
```
Comment (bcrochet) [inline .vscode/settings.json:3]: I don't think we want this.

**Review by bcrochet — CHANGES_REQUESTED** [review]
Remove .vscode/settings.json

**Review by bcrochet — APPROVED** [review]
/lgtm

**Review by mrhillsman — APPROVED** [review]
